@here/harp-mapview-decoder/lib/TileLoader.ts

```diff
@@ -104,25 +104,19 @@ export class TileLoader {
     loadAndDecode(): Promise<TileLoaderState> {
         switch (this.state) {
             case TileLoaderState.Loading:
+            case TileLoaderState.Loaded:
+            case TileLoaderState.Decoding:
+                // tile is already loading
                 this.countRequests++;
                 return this.donePromise!;
 
             case TileLoaderState.Ready:
             case TileLoaderState.Failed:
             case TileLoaderState.Initialized:
             case TileLoaderState.Canceled:
+                // restart loading
                 this.countRequests++;
-                this.ensureLoadingStarted();
-                return this.donePromise!;
-
-            case TileLoaderState.Loaded:
-                this.countRequests++;
-                this.startDecodeTile();
-                return this.donePromise!;
-
-            case TileLoaderState.Decoding:
-                this.cancelDecoding();
```

Comment:
I'm not sure this is exactly correct, if you are decoding a Tile, and then call loadAndDecode, it probably still makes sense to cancel the request. For example, if you are changing styles, and you don't cancel the current request, you will get tiles with different styles.

Reply:
When the style is changed the tiles of a datasource are marked as dirty and evicted from the cache. This also cancels running requests. There still seems to be a bug where tiles are displayed with different styles but it can also be triggered with the changes in this pull request.

Reply:
You're right, ok cool.